### PR TITLE
[LibraryInfo] Add new LibraryInfo component (attempt #2)

### PR DESCRIPTION
**Pull request by ajsecord**
This is identical to #2401, my git history got badly munged, so this is just a fresh copy.

### CODE REVIEW

**Review by jverkoey — APPROVED** [review]

**Comment by ianegordon** [comment]
Please merge when Travis is happy.

File: components/LibraryInfo/tests/unit/LibraryInfoTests.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  func testVersionFormat() {
+    // Given
+
+    // This regex pattern does the following:
+    // Accept: "42.0.0", etc.
+    // Reject: "0.0.0", "1.2", "1", "-1.2.3", "Hi, I'm a version 1.2.3", "1.2.3 is my version", etc.
+    //
+    // Note the major version must be >= 1 since "0.0.0" is used as the version when something goes
+    // wrong in the underlying code and we're in Release mode.
+    let pattern = "^[1-9]+\\.[0-9]+\\.[0-9]+$"
```
Comment (ianegordon) [inline components/LibraryInfo/tests/unit/LibraryInfoTests.swift:31]: Is this string intended to be Semantic Version compliant.  If so, we may have 1.2.3-alpha, 4.5.6-rc, 5.4.3-rc.99.

**Review by ianegordon — APPROVED** [review]
Please see note about semantic versioning.  If it doesn't apply, we are set.

File: components/LibraryInfo/tests/unit/LibraryInfoTests.swift (hunk 1 above)
Comment (ajsecord) [reply components/LibraryInfo/tests/unit/LibraryInfoTests.swift:31]: Oooh, or 1.2.3-alpha+buildnum, too. 

No, this is intended to test that we're returning our API in the header, which just specifies "X.Y.Z". If we change that then we can update this test.

**Pull request merged**